Update Page Not Found view to reflect new v1 designs

Fabricevladimir opened the pull request:
Closes #117

will-nemo commented on src/style-sheets/modules/_colors.scss:
```diff
@@ -10,6 +10,7 @@ $olLTAccent: #43a850;
 $olLTHeader: #262626;
 $olNavLink: #232936;
 $olLTCopy: #2b2d31;
+$olLightCopy: #777777;
```
so the variable before this is olLightThemeCopy. Is that different than olLightCopy?

Fabricevladimir replied:
There really isn't I was just struggling to come up with a variable name! Suggestions?

will-nemo replied:
What about changing the name of both variables. Maybe olLTSolidCopy and olLTLightCopy or olLTPrimaryCopy and olLTSecondaryCopy?

will-nemo commented on src/style-sheets/pages/_page-not-found-view.scss:
```diff
@@ -1,17 +1,86 @@
+
+      & p {
+        color: $olLightCopy;
+        font-size: $olH5Size;
```
if we're using the size of an h5 should be just be using an h5 tag instead? I'd say if we were gonna control the size of a p tag in the module then we should be using rem for that.

Fabricevladimir replied:
No, I don't think that should be a header.

will-nemo replied:
then if we're going to be changing the size in the module then we should be using rem instead of em right? https://css-tricks.com/rem-global-em-local/

will-nemo commented:
Also @dlinb (Chum) made an svg with animation in it that we can use for this view. Probably should be using that

will-nemo commented:
Since there's not a lot of content on that view it seems to have a ton of white space underneath the footer. Is there anyway to fix that? Maybe more margin above and below the container to fill the page?

will-nemo commented:
<img width="1448" alt="Screen Shot 2020-03-27 at 10 13 35 PM" src="https://user-images.githubusercontent.com/30677661/77913881-0934b780-7263-11ea-92b4-717b9d705469.png">

Also when the view is in mobile is seems that the font stays the same size. It looks like in the figma it gets smaller. This can be fixed by the rem maybe or it would be fixed by changing the font size in the media query.

Fabricevladimir commented:
I'm not sure what you mean. The font stays constant at a whopping 24px on figma
> Also when the view is in mobile is seems that the font stays the same size. It looks like in the figma it gets smaller. This can be fixed by the rem maybe or it would be fixed by changing the font size in the media query.

Fabricevladimir commented:
> Also @dlinb (Chum) made an svg with animation in it that we can use for this view. Probably should be using that

Can you point me to that?

Fabricevladimir commented:
> Since there's not a lot of content on that view it seems to have a ton of white space underneath the footer. Is there anyway to fix that? Maybe more margin above and below the container to fill the page?

Underneath? Or do you mean on top of the footer? I agree, though; there is a lot of white space, but that's the way it is on figma. Whether it was intended to be like that, I'm not sure.

will-nemo commented:
> I'm not sure what you mean. The font stays constant at a whopping 24px on figma
> 
> > Also when the view is in mobile is seems that the font stays the same size. It looks like in the figma it gets smaller. This can be fixed by the rem maybe or it would be fixed by changing the font size in the media query.

Ah I see you're right. It seems to be the container containing the copy has margin auto which is centering it in the container and making the width of the p tag small. You could get rid of the margins if you made the .page-not-found-view-text-container a flex column and centered the text with that. 

That won't work though since the container holding the image and copy is a flex row even in mobile view. If we changed it to a column in mobile view then that would work.

Fabricevladimir commented:
> Ah I see you're right. It seems to be the container containing the copy has margin auto which is centering it in the container and making the width of the p tag small. You could get rid of the margins if you made the .page-not-found-view-text-container a flex column and centered the text with that.
> 
> That won't work though since the container holding the image and copy is a flex row even in mobile view. If we changed it to a column in mobile view then that would work.

Yeah, on mobile it's supposed to be centered apparently. Also, in this case, the container flex direction shouldn't have mattered because the size is forcing the content to wrap. I've reworked it to make it simpler. Fixes should be easier now.

will-nemo commented:
> <img alt="Screen Shot 2020-03-30 at 8 45 28 AM" width="375" src="https://user-images.githubusercontent.com/30677661/77978384-fd330f00-72cf-11ea-9fba-ca910d30a236.png">
> 
> Also when the view is in mobile is seems that the font stays the same size. It looks like in the figma it gets smaller. This can be fixed by the rem maybe or it would be fixed by changing the font size in the media query.

i added the wrong image earlier lol

will-nemo commented:
<img width="335" alt="Screen Shot 2020-03-30 at 9 49 05 PM" src="https://user-images.githubusercontent.com/30677661/77978498-5602a780-72d0-11ea-93fb-b5722ecfe52b.png">


https://www.figma.com/file/zukLdUXgDYBhPAU8NZbljW/OL-Website-Workspace?node-id=1633%3A174

Ya text body width looks way smaller than the mock up copy. Button also doesn't seem centered

Fabricevladimir commented:
> Ya text body width looks way smaller than the mock up copy. Button also doesn't seem centered. That's at 320px view port like the xsmall screen is designed for

Fabricevladimir commented:
> Ya text body width looks way smaller than the mock up copy. Button also doesn't seem centered. That's at 320px view port like the xsmall screen is designed for
Nice catch. I think it's the margin: auto that skewed it. The links in footer seem to be slightly off as well actually

will-nemo commented:
> > Ya text body width looks way smaller than the mock up copy. Button also doesn't seem centered. That's at 320px view port like the xsmall screen is designed for
> 
> Nice catch. I think it's the margin: auto that skewed it. The links in footer seem to be slightly off as well actually

Ya they are, I have that fixed in my contact us PR

Fabricevladimir commented:
I played around with the margins between the paragraphs a bit. Let me know what you think.

will-nemo commented on src/pages/PageNotFoundView.jsx:
```diff
@@ -9,13 +12,20 @@ export default function PageNotFoundView() {
+          <p>
+            404 Page not found, <strong>{location.pathname}</strong>
+          </p>
+          <p>Let's hurry; there is nothing to fear here</p>
```
nit picky but there should be a period at the end of this

Fabricevladimir replied:
You are being nit picky! All good, you're right.

will-nemo commented on src/pages/PageNotFoundView.jsx:
```diff
@@ -9,13 +12,20 @@ export default function PageNotFoundView() {
+          </p>
+          <p>Let's hurry; there is nothing to fear here</p>
+          <Button>
+            <Link to='/'>Take me home</Link>
```
The button should be wrapped in a link so we don't have to click the text to be linked back home. 
```
          <Link to='/'>
            <Button>
              <span>Take me home</span>
            </Button>
          </Link>
```

will-nemo commented on src/style-sheets/pages/_page-not-found-view.scss:
```diff
@@ -1,17 +1,110 @@
+        line-height: 1.56rem;
+
+        text-align: center;
+        font-weight: 600;
```
I know the font weight in figma says 600 but it looks so much bolder than the mock up. This will be fine for now

will-nemo commented:
Looks like some merge conflicts, after that's fixed should be good to go.
